src/libs/DateUtils.js

```diff
@@ -291,6 +291,105 @@ function getDateStringFromISOTimestamp(isoTimestamp) {
     return dateString;
 }
 
+/**
+ * @returns {string} example: 2023-05-16
+ */
+function getThirtyMinutesFromNow() {
+    return moment().add(30, 'minutes').format('YYYY-MM-DD HH:mm:ss');
+}
+
+function getOneHourFromNow() {
+    return moment().add(1, 'hours').format('YYYY-MM-DD HH:mm:ss');
+}
+
+/**
+ * @returns {string} example: 2023-05-16
+ */
+function getEndOfToday() {
+    return moment().endOf('day').format('YYYY-MM-DD HH:mm:ss');
+}
+
+/**
+ * @returns {string} example: 2023-05-16
+ */
+function getEndOfWeekFromNow() {
+    return moment().add(7, 'days').format('YYYY-MM-DD HH:mm:ss');
+}
+
+/**
+ * @param {string} time
+ * @returns {string} example: 2023-05-16
+ * it's kind of 'newer' date
+ */
+function extractDate(time) {
+    return moment(time).format('YYYY-MM-DD');
+}
+
+/**
+ * @param {string} dateTimeString
+ * @returns {string} example: 2023-05-16
+ * it's kind of 'newer' date
+ */
+function extractTime(dateTimeString) {
+    return moment(dateTimeString).format('hh:mm');
+}
+/**
+ * @param {string} dateTimeString
+ * @returns {string} example: 2023-05-16
+ * it's kind of 'newer' date
+ */
+function extractTime12Hour(dateTimeString) {
+    return moment(dateTimeString).format('hh:mm A');
+}
+
+/**
+ * @param {string} dateTimeString
+ * @returns {string} example: 2023-05-16
+ * it's kind of 'newer' date
+ */
+function formatDateTo12Hour(dateTimeString) {
+    if (!dateTimeString) return '';
+    return moment(dateTimeString).format('YYYY-MM-DD hh:mm A');
+}
+
+/**
+ * @param {string} type - one of the values from CONST.CUSTOM_STATUS_TYPES
+ * @returns {string} example: 2023-05-16
+ */
+function getDateBasedFromType(type) {
+    switch (type) {
+        case CONST.CUSTOM_STATUS_TYPES.THIRTY_MINUTES:
+            return getThirtyMinutesFromNow();
+        case CONST.CUSTOM_STATUS_TYPES.ONE_HOUR:
+            return getOneHourFromNow();
+        case CONST.CUSTOM_STATUS_TYPES.AFTER_TODAY:
+            return getEndOfToday();
+        case CONST.CUSTOM_STATUS_TYPES.AFTER_WEEK:
+            return getEndOfWeekFromNow();
+        case CONST.CUSTOM_STATUS_TYPES.NEVER:
+            return CONST.CUSTOM_STATUS_TYPES.NEVER;
+        default:
+            return '';
+    }
+}
+
+/**
+ * @param {string} data - one of the values from CONST.CUSTOM_STATUS_TYPES or data in format YYYY-MM-DD HH:mm
+ * @returns {string} example: 2023-05-16 11:10 PM or 'Today'
+ */
+function getDateForTitleBasedFromType(data) {
```

Comment:
The function name isn't as descriptive as it could be - since it accepts things other than just the type. Similar to above I'd also make it specific to status or `ClearStatus`.

Reply:
`clearAfter` always has a predefined value. So in case we have no value, it means we have type `never`. I'll update the naming to `getLocalizedTimePeriodDescription`